sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -937,21 +960,6 @@ static void _recv_DIO_for_existing_dodag(gnrc_rpl_instance_t *inst, gnrc_rpl_dio
     }
 #endif
 
-    if (GNRC_RPL_COUNTER_GREATER_THAN(dio->version_number, dodag->version)) {
-        if (dodag->node_status == GNRC_RPL_ROOT_NODE) {
-            dodag->version = GNRC_RPL_COUNTER_INCREMENT(dio->version_number);
-            trickle_reset_timer(&dodag->trickle);
-        }
-        else {
-            dodag->version = dio->version_number;
-            gnrc_rpl_local_repair(dodag);
-        }
-    }
-    else if (GNRC_RPL_COUNTER_GREATER_THAN(dodag->version, dio->version_number)) {
-        trickle_reset_timer(&dodag->trickle);
-        return;
-    }
```

Comment:
I think, different DODAG versions should be treated as different DODAGs because they affect also our parent set etc. (e.g., per RFC all parents in the parent set must be the same DODAG version). So the old logic of just increasing our version and leaving the rest as it is wasn't really RFC conformant. 
Is my understanding correct?

Reply:
> The version number is incremented when something happens that requires that the DODAG be rebuilt. It's not a different DODAG. One should attach to the highest version number (with lollipop numbering, I think)

Yes, that was also my understanding.
But implementation wise, I was thinking that we could still treat a new DODAG version the same as we treat joining a completely different DODAG: clear the parent set, reset trickle, reset other parameters... 
The old implementation effectively also just triggered a local repair (=> implemented as poising of all ranks and cleanup of the current dodag).

But thinking about it again now, we probably don't want to reset all parameters, like dtsn or DAO sequence.